packages/orama/src/trees/avl.ts

```diff
@@ -198,121 +237,161 @@ export function create<K, V>(key: K, value: V): Node<K, V> {
-    return node
+      parent.right = newNode;
   }
 
-  const balanceFactor = getBalanceFactor(node)
-
-  if (balanceFactor === BALANCE_STATE.UNBALANCED_LEFT) {
-    if (key < (node.left as Node<K, V>).key) {
-      node = rotateRight(node)
-    } else {
-      node.left = rotateLeft(node.left as Node<K, V>)
-      node = rotateRight(node)
+  current = newNode;
+
+  while (parent) {
+    const balanceFactor = getBalanceFactor(parent);
+    
+    if (balanceFactor === BALANCE_STATE.UNBALANCED_LEFT) {
+        if (key < (parent.left as Node<K, V>).key) {
+            parent = rotateRight(parent);
+        } else {
+            parent.left = rotateLeft(parent.left as Node<K, V>);
+            parent = rotateRight(parent);
+        }
     }
-  }
 
-  if (balanceFactor === BALANCE_STATE.UNBALANCED_RIGHT) {
-    if (key > (node.right as Node<K, V>).key) {
-      node = rotateLeft(node)
-    } else {
-      node.right = rotateRight(node.right as Node<K, V>)
-      node = rotateLeft(node)
+    if (balanceFactor === BALANCE_STATE.UNBALANCED_RIGHT) {
+        if (key > (parent.right as Node<K, V>).key) {
+            parent = rotateLeft(parent);
+        } else {
+            parent.right = rotateRight(parent.right as Node<K, V>);
+            parent = rotateLeft(parent);
+        }
     }
-  }
 
-  return node
-}
+    if (parent == root) {
+        break;
+    }
 
-export function find<K, V>(node: Node<K, V>, key: K): V | null {
-  if (!node) {
-    return null
+    current = parent;
+    parent = getNodeParent(root, current.key);
   }
 
-  if (node.key === key) {
-    return node.value
-  }
+  return root;
+}
 
-  if (key < node.key) {
-    return node.left ? find(node.left, key) : null
-  }
+function getNodeParent<K, V>(root: Node<K, V>, key: K): Node<K, V> | null {
+    let current = root;
+    let parent = null;
+
+    while (current !== null) {
+        if (key < current.key) {
+            parent = current;
+            current = current.left as Node<K, V>;
+        } else if (key > current.key) {
+            parent = current;
+            current = current.right as Node<K, V>;
+        } else {
+            break;
+        }
+    }
 
-  return node.right ? find(node.right, key) : null
+    return parent;
 }
 
-export function remove<K, V>(node: Node<K, V>, key: K): Node<K, V> | null {
-  if (!node) {
-    return null
-  }
+export function find<K, V>(root: Node<K, V>, key: K): V | null {
+  let node = root;
 
-  if (key < node.key) {
-    node.left = remove(node.left as Node<K, V>, key)
-  } else if (key > node.key) {
-    node.right = remove(node.right as Node<K, V>, key)
-  } else {
-    if (!node.left && !node.right) {
-      return null
+  while (node) {
+    if (key < node.key) {
+      node = node.left as Node<K, V>;
+    } else if (key > node.key) {
+      node = node.right as Node<K, V>;
+    } else {
+      return node.value;
     }
+  }
 
-    if (!node.left) {
-      return node.right as Node<K, V>
-    }
+  return null;
```

Comment:
should we use getNodeByKey here ?

Reply:
Not sure I follow?

Reply:
@allevo as discussed privately, we'll open a separate PR if there's anything we can improve